applications/StructuralMechanicsApplication/custom_advanced_constitutive/multi_linear_elastic_1d_law.h

```diff
@@ -0,0 +1,146 @@
+//
+//  Main authors:    Klaus B. Sautter
+//
+
+#if !defined (KRATOS_MULTI_LINEAR_ELASTIC_1D_LAW_H_INCLUDED)
+#define  KRATOS_MULTI_LINEAR_ELASTIC_1D_LAW_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "custom_constitutive/truss_constitutive_law.h"
+
+namespace Kratos
+{
+
+    /**
+     * @namespace MultiLinearElastic1DLaw
+     *
+     * @brief This constitutive law represents a multi linear elastic 1d claw
+     *
+     * parameters to define the strain-energy function are
+     * MULTI_LINEAR_ELASTICITY_MODULI
+     * MULTI_LINEAR_ELASTICITY_STRAINS
+     *
+     * Johannes Linhard - Numerisch-mechanische Betrachtung des Entwurfsprozesses von Membrantragwerken
+     *
+     * @author Klaus B Sautter
+     */
+
+
+    class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) MultiLinearElastic1DLaw : public TrussConstitutiveLaw
+    {
+    public:
+
+        KRATOS_CLASS_POINTER_DEFINITION( MultiLinearElastic1DLaw );
+
+        /**
+         * Life Cycle
+         */
+
+        /**
+         * Default constructor.
+         */
+        MultiLinearElastic1DLaw();
+
+        ConstitutiveLaw::Pointer Clone() const override;
+
+        /**
+         * Copy constructor.
+         */
+        MultiLinearElastic1DLaw (const MultiLinearElastic1DLaw& rOther);
+
+
+        /**
+         * Destructor.
+         */
+        ~MultiLinearElastic1DLaw() override;
+
+        /**
+         * This function provides the place to perform checks on the completeness of the input.
+         * It is designed to be called only once (or anyway, not often) typically at the beginning
+         * of the calculations, so to verify that nothing is missing from the input
+         * or that no common error is found.
+         * @param rMaterialProperties: The properties of the material
+         * @param rElementGeometry: The geometry of the element
+         * @param rCurrentProcessInfo: The current process info instance
+         */
+        int Check(
+            const Properties& rMaterialProperties,
+            const GeometryType& rElementGeometry,
+            const ProcessInfo& rCurrentProcessInfo
+        ) override;
+
+        double& CalculateValue(ConstitutiveLaw::Parameters& rParameterValues,
+            const Variable<double>& rThisVariable,double& rValue) override;
+
+    protected:
+
+        ///@name Protected static Member Variables
+        ///@{
+
+        ///@}
+        ///@name Protected member Variables
+        ///@{
+
+        ///@}
+        ///@name Protected Operators
+        ///@{
+
+        ///@}
+        ///@name Protected Operations
+        ///@{
+        ///@}
+
+    private:
+
+        ///@name Static Member Variables
+        ///@{
+
+        ///@}
+        ///@name Member Variables
+        ///@{
+        ///@}
+        ///@name Private Operators
+        ///@{
+
+        ///@}
+        ///@name Private Operations
+        ///@{
+        ///@}
+
+
+        ///@}
+        ///@name Private  Access
```

Comment:
I personally wouldn't put all the commented stuff. It's just scrolling time ;)

Reply:
In Structural mech we usually do it just in case we have to add something in the future :D

Reply:
I would say we did that so far

but as @tteschemacher says this adds a lot of unnecessary lines
I also remove it typically
I would argue the other way round, it can be added later if necessary
But this is really a minor minor issue :D